### PR TITLE
Gets correct cluster from datacenter

### PR DESCRIPTION
**Pull request by orrabin**
This fixes http://projects.theforeman.org/issues/23815

### CODE REVIEW

File: lib/fog/ovirt/requests/compute/v4/create_vm.rb [hunk 1]
```diff
@@ -27,9 +27,15 @@ def process_vm_opts(opts)
           # rubocop:disable Metrics/AbcSize, Metrics/CyclomaticComplexity, Metrics/PerceivedComplexity
           def create_vm(attrs)
             attrs = attrs.dup
-            attrs[:cluster_name] ||= datacenter.clusters.first.name unless attrs[:cluster]
+
+            unless attrs[:cluster]
+              cluster = client.system_service.clusters_service.list(:search => "datacenter=#{datacenter_hash[:name]}").first
+              attrs[:cluster] = cluster
+              attrs[:cluster_name] ||= cluster.name
```
Comment (adamruzicka) [inline lib/fog/ovirt/requests/compute/v4/create_vm.rb]: Can there be a situation where `attrs[:cluster]` is unset, but `attrs[:cluster_name]` is set?

Comment (orrabin) [reply lib/fog/ovirt/requests/compute/v4/create_vm.rb]: Not that I've seen so far but it's possible if someone calls `create_vm` with the cluster name.
However this should probably change - if there is a name we should search by name and not datacenter and it we searched by datacenter the name should follow.

**Comment by orrabin** [comment]
@adamruzicka thanks, I changed this to include an option for a provided `cluster_name` with no id.

**Review by adamruzicka — CHANGES_REQUESTED** [review]
Looks good to me but :policeman: is not happy

**Comment by orrabin** [comment]
@adamruzicka there were too many lines, I noticed we had an `if` with only 1 statement in it so it was a good opportunity to refactor that as well.

**Comment by orrabin** [comment]
@adamruzicka rubocop was failing on an unrelated file this time, I'm fixing here but this will causes failures in other PRs as well.

**Review by adamruzicka — APPROVED** [review]

**Comment by adamruzicka** [comment]
Thanks @orrabin !

**Pull request merged**